### PR TITLE
Add Imdb 8 digit

### PR DESCRIPTION
**Pull request by clinton-hall**
### Description of what this fixes:

Added support for 8-digit imdb id.

### Related issues:
#7308

### CODE REVIEW

File: couchpotato/core/plugins/userscript/template.js_tmpl [hunk 1]
```diff
@@ -100,7 +100,7 @@ var osd = function(){
 
         // Try and get imdb url
         try {
-            var regex = new RegExp(/tt(\d{7})/);
+            var regex = new RegExp(/tt(\d{8})/);
```
Comment (DisasteR) [inline couchpotato/core/plugins/userscript/template.js_tmpl]: I'm not sure about using a fixed 8 size value.
Why not just validate digits or using same pattern as python ?

File: couchpotato/core/helpers/variable.py [hunk 2]
```diff
@@ -194,7 +194,7 @@ def getImdb(txt, check_inside = False, multiple = False):
         output.close()
 
     try:
-        ids = re.findall('(tt\d{4,7})', txt)
+        ids = re.findall('(tt\d{4,8})', txt)
 
         if multiple:
             return removeDuplicate(['tt%07d' % tryInt(x[2:]) for x in ids]) if len(ids) > 0 else []
```
Comment (DisasteR) [inline couchpotato/core/helpers/variable.py]: You forgot this one. In the formater

**Comment by DisasteR** [comment]
Looks like imdb accept any number of leading 0 in the id's. So why not just removing the limit.

**Comment by DisasteR** [comment]
@RuudBurger what about the missing changes ?

**Comment by boredazfcuk** [comment]
Hi, I've just built a new container from the latest develop branch but I'm seeing this:
`
HTTPError: 404 Client Error: Not Found for url: https://api.couchpota.to/info/tt03513498/?ignore=1
`
a lot in the logs (every time it refreshes the suggestions).

Are my errors due to the api.couchpotato.to code not being able to handle the 8-digit tt codes being sent to it?

**Comment by mdklapwijk** [comment]
Same kind of goes for torrentapi.org e.g.:
- https://torrentapi.org/pubapi_v2.php?token=s41lhk2uep&mode=search&search_imdb=tt05073642&app_id=couchpotato
- https://torrentapi.org/pubapi_v2.php?token=s41lhk2uep&mode=search&search_imdb=tt5073642&app_id=couchpotato

**Comment by horganhoozen** [comment]
So after this update I am unable to use YTS as a provider. The query (ran manually) comes back with no results. If I remove one of the leading zeros in the imdb id number (to make it only 7 digits), the query, ran manually, comes back with results.

**Comment by mdklapwijk** [comment]
#7318

**Pull request merged**